bgpd/bgpd.c

```diff
@@ -2575,9 +2582,11 @@ int peer_group_delete(struct peer_group *group)
 
 	for (ALL_LIST_ELEMENTS(group->peer, node, nnode, peer)) {
 		other = peer->doppelganger;
+		peer_notify_unconfig(peer);
```

Comment:
We should avoid putting the changes in peer_group_delete() as it is called from other places. Instead we should call peer_notify_unconfig() judiciously at the places peer_group_delete() is called.

Reply:
I think it's safe to do this unconditionally because peer_group_delete() is used in bgp_delete() only. Of course, I could move this out there, but don't see the reason why. Could you explain me the issues you see?

Reply:
@ton31337 

I would request for moving the notification sending wrapper to be called from the place peer_group_delete() is being called for the following reasons : 

1. peer_group_delete() gets also called from bgp_delete(). Inside bgp_delete() there is peer loop where we are sending BGP_NOTIFY_CEASE_ADMIN_SHUTDOWN notification. Later in the same function we call peer_group_delete() were will be sending different notification. 

2. It would be nice to see constancy while sending notification b/w individual peers (peer_delete()) and peer group (peer_group_delete()). 
I see advantages behind doing so, as we never know never know if peer_group_delete() gets called from some other hook function in future, from where we might need to send some other notification. So rather than doing cleanup of peer_group_delete() in future, if we design properly now, do no housing keeping later :)

Reply:
Fixed